pagopa/io-web-profile
# [IOPID-1008] : Cookie Banner

## Short description

The purpose of this PR is to integrate the cookie banner provided by OneTrust.

## List of changes proposed in this pull request

- Adding OneTrust SDK
- OneTrust integration for the cookie banner
- Update Readme
- Adding FIX ME Comments

## How to test

Insert the following variable into the local environment file:

```NEXT_PUBLIC_DEV_MODE=true```

Please note that this variable will be omitted when building the production package.
This is necessary to determine if the application is in local development mode or production. 
It's required because in production the OT NOTICE ID (OneTrust) it's different

Start the project and ensure that the cookie banner is displayed.
The choice will be saved within OneTrust's cookies. (OptanonConsent -> Group C0002) 
If we have the consence, it will also grant permission for tracking activities in Mixpanel; otherwise, Mixpanel must be restricted.
